### PR TITLE
Thread view integration with HomeServer API

### PR DESCRIPTION
**Pull request by gsouquet**
Partially addresses https://github.com/vector-im/element-web/issues/19588 and https://github.com/vector-im/element-web/issues/19961

Integrates with the `/relations` API and the bundled relationships
Also provides a fallback for when a server does not have support for those APIs

- Integrate threads with homeserver APIs
- Store pagination token directly on timeline
- Update for threads with server side support

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/@types/event.ts [hunk 1]
```diff
@@ -114,6 +115,12 @@ export enum MsgType {
     KeyVerificationRequest = "m.key.verification.request",
 }
 
+export interface ThreadBundledRelation {
```
Comment (t3chguy) [inline src/@types/event.ts:118]: We normally prefix our interfaces with an `I`

File: src/sync.ts [hunk 2]
```diff
@@ -1704,8 +1710,8 @@ export class SyncApi {
     /**
      * @experimental
      */
-    private processThreadEvents(room: Room, threadedEvents: MatrixEvent[]): void {
-        return this.client.processThreadEvents(room, threadedEvents);
+    private processThreadEvents(room: Room, threadedEvents: MatrixEvent[], opts): void {
```
Comment (t3chguy) [inline src/sync.ts:1713]: type?

**Review by t3chguy — APPROVED** [review]
Looks sane otherwise

**Comment by gsouquet** [comment]
Asking for a re-review as some things to do with back pagination have significantly changed since the last review

**Comment by gsouquet** [comment]
Superseded by https://github.com/matrix-org/matrix-js-sdk/pull/2142

**Pull request closed without merging**